designer/app.py

```diff
@@ -1015,6 +1016,7 @@ def action_btn_save_pressed(self, exit_on_save=False, *args):
         else:
             self.save_project()
             if exit_on_save:
+                self.remove_temp_proj_directories()
```

Comment:
instead of calling `self.remove_temp_proj_directories()` in multiple locations, let's call it inside `_perform_quit`. So you can remove all calls and just use it on perform_quit